WalletWasabi.Backend/Controllers/APNTokensController.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using WalletWasabi.Backend.Data;
+using WalletWasabi.Backend.Models;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Backend.Controllers
+{
+	[Route("api/v" + Constants.BackendMajorVersion + "/btc/[controller]")]
+	[ApiController]
+	[Produces("application/json")]
+	public class APNTokensController : ControllerBase
+	{
+		private readonly APNTokensContext Db;
+
+		public APNTokensController(APNTokensContext db)
+		{
+			Db = db;
+		}
+
+		[HttpPost]
+		[ProducesResponseType(200)]
+		[ProducesResponseType(400)]
+		public async Task<IActionResult> StoreTokenAsync([FromBody] AppleDeviceToken token)
```

Comment:
You need some sort of spam protection here. This is a ddos attack vector.  See https://github.com/NicolasDorier/NicolasDorier.RateLimits for a basic limiter but you should also add some sort of interactive verification because an asshole will spam your db

Reply:
Wasabi suffered [DDoS in June](https://blog.wasabiwallet.io/wve-006-ddos-attack-report/) so I agree that there is a real threat to defend against. At the same time, to coordinate peers in a 30 second window we have to process hundreds / thousands of requests. I'm not sure what the threshold of stability is here and probably won't until we go out try & have things break. By interactive verification, do you mean issuing tokens to authorize these requests?

Reply:
Yes, make every device sending you these requests require some form of auth (eg store a pubkey on backend and have them sign the HTTP body.)